src/lib.rs

```diff
@@ -541,64 +546,37 @@ fn check_json_errors(error: reqwest::Error, api: &str) {
     println!("An error ❌ occurred while parsing the JSON obtained from the {} API. Error description: {}.", &api, error.description())
 }
 
-pub fn read_from_file(file: &str, with_ip: &str, with_output: &str) {
+pub fn read_from_file(file: &str, with_ip: &str, with_output: &str) -> Result<()> {
     match File::open(&file) {
         Ok(f) => {
             let f = BufReader::new(f);
             for domain in f.lines() {
                 let domain = domain.unwrap().to_string();
                 let file_name = [&domain, ".txt"].concat();
-                get_subdomains(&domain, &with_ip, &with_output, &file_name)
+                get_subdomains(&domain, &with_ip, &with_output, &file_name)?;
             }
         }
         Err(e) => {
             println!("Can't open file 📁 {}. Error: {}", &file, e.description());
         }
     }
+    Ok(())
 }
 
-fn write_to_file(data: &str, subdomain_ip: &str, file_name: &str, with_ip: &str) {
-    if with_ip == "y" {
-        let data = &[data, ",", subdomain_ip, "\n"].concat();
-        if Path::new(&file_name).exists() {
-            let mut output_file = OpenOptions::new()
-                .append(true)
-                .open(&file_name)
-                .expect("Can't open file.");
-            output_file
-                .write_all(&data.as_bytes())
-                .expect("Failed writing to file.");
-        } else {
-            File::create(&file_name).expect("Failed to create file.");
-            let mut output_file = OpenOptions::new()
-                .append(true)
-                .open(&file_name)
-                .expect("Can't open file.");
-            output_file
-                .write_all(&data.as_bytes())
-                .expect("Failed writing to file.");
-        }
+fn write_to_file(data: &str, subdomain_ip: &str, file_name: &str, with_ip: &str) -> Result<()> {
+    let data = if with_ip == "y" {
+        [data, ",", subdomain_ip, "\n"].concat()
     } else {
-        let data = &[data, "\n"].concat();
-        if Path::new(&file_name).exists() {
-            let mut output_file = OpenOptions::new()
-                .append(true)
-                .open(&file_name)
-                .expect("Can't open file.");
-            output_file
-                .write_all(&data.as_bytes())
-                .expect("Failed writing to file.");
-        } else {
-            File::create(&file_name).expect("Failed to create file.");
-            let mut output_file = OpenOptions::new()
-                .append(true)
-                .open(&file_name)
-                .expect("Can't open file.");
-            output_file
-                .write_all(&data.as_bytes())
-                .expect("Failed writing to file.");
-        }
-    }
+        [data, "\n"].concat()
+    };
+    let mut output_file = OpenOptions::new()
+        .append(true)
+        .create(true)
+        .open(&file_name)
+        .context("Can't open file")?;
```

Comment:
I've sticked to the original error message here, you can make it more verbose with:
```
.with_context(|_| format!("Can't open file: {:?}", &file_name))?;
```

Reply:
Please can you include it in your branch to merge directly?